hudi-common/src/main/java/org/apache/hudi/common/fs/ConsistencyGuardConfig.java

```diff
@@ -30,29 +31,42 @@
  */
 public class ConsistencyGuardConfig extends DefaultHoodieConfig {
 
-  private static final String CONSISTENCY_CHECK_ENABLED_PROP = "hoodie.consistency.check.enabled";
-  private static final String DEFAULT_CONSISTENCY_CHECK_ENABLED = "false";
-
   // time between successive attempts to ensure written data's metadata is consistent on storage
-  private static final String INITIAL_CONSISTENCY_CHECK_INTERVAL_MS_PROP =
-      "hoodie.consistency.check.initial_interval_ms";
-  private static long DEFAULT_INITIAL_CONSISTENCY_CHECK_INTERVAL_MS = 400L;
+  public static final ConfigOption<String> CONSISTENCY_CHECK_ENABLED_PROP = ConfigOption
+      .key("hoodie.consistency.check.enabled")
+      .defaultValue("false")
+      .withDescription("Should HoodieWriteClient perform additional checks to ensure written files’ are listable "
+          + "on the underlying filesystem/storage. Set this to true, to workaround S3’s eventual consistency model "
+          + "and ensure all data written as a part of a commit is faithfully available for queries.");
```

Comment:
Just mention that this property is to handle S3 eventual consistency, and say that this property is no longer required to be used since S3 is now strongly consistent. Also mention that this will be removed in future releases.